### ISSUE TITLE
Ride Status not available with ObjC ?

### ISSUE THREAD

**Issue by fvisticot** [uber/rides-ios-sdk#94]
UBSDKRide does not expose ride Status.
How to get the ride status after a call to fetchRideDetails ?

``` Objective-C
@interface UBSDKRide : NSObject

/// Contains the information about the destination of the trip, if one has been set.
@property (nonatomic, readonly, strong) UBSDKRideRequestLocation * _Nullable destination;

/// The object that contains driver details. Only non-null during an ongoing trip.
@property (nonatomic, readonly, strong) UBSDKDriver * _Nullable driver;

/// The object that contains the location information of the vehicle and driver.
@property (nonatomic, readonly, strong) UBSDKRideRequestLocation * _Nullable driverLocation;

/// The estimated time of vehicle arrival in minutes.
@property (nonatomic, readonly) NSInteger eta;

/// The object containing the information about the pickup for the trip.
@property (nonatomic, readonly, strong) UBSDKRideRequestLocation * _Nullable pickup;

/// The unique ID of the Request.
@property (nonatomic, readonly, copy) NSString * _Nullable requestID;

/// The surge pricing multiplier used to calculate the increased price of a Request.
@property (nonatomic, readonly) double surgeMultiplier;

/// The object that contains vehicle details. Only non-null during an ongoing trip.
@property (nonatomic, readonly, strong) UBSDKVehicle * _Nullable vehicle;

@end
```

**Comment by jbrophy17** [comment]
You are correct, good find. This is because Objective-C does not support an optional Enum (which is how rideStatus is defined). The fix will be not having rideStatus optional and instead giving it a default value.

If you like, since you found this, you can put up a PR and I would be happy to review it

**Comment by fvisticot** [comment]
Can you please explain a little bit more where to do the modification ? can we add this status property in the UBSDKRide class ? How this property will be updated by current API ? Tx for helping

**Comment by jbrophy17** [comment]
The status property exists, it just is not currently being exposed in the Objective-C interface. In order to expose it, it needs to be something that can be represented in Objective-C. So, what I think should work is a slight modification in `Ride.swift`

Need to change:
```swift
    /// The status of the Request indicating state.
    public private(set) var status: RideStatus?
```
to:
```swift
    /// The status of the Request indicating state.
    public private(set) var status: RideStatus = .Unknown
```
and make adjustments wherever `status` has been used

**Comment by jbrophy17** [comment]
Fixed in [v0.5.3](https://github.com/uber/rides-ios-sdk/releases/tag/v0.5.3)
